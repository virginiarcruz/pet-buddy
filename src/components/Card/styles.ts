import styled, { css } from 'styled-components'

export const CardContainer = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.color.lighterGray};
    border-radius: ${theme.border.radius.medium};
    padding: 16px;
  `}
`
export const Header = styled.label`
  display: flex;
  align-items: center;
  cursor: pointer;
`

export const Input = styled.input`
  display: none;

  &:checked {
    & + label {
      .icon-button {
        transform: rotate(180deg);
      }
    }
    & + label + .card-content {
      max-height: 100vh;
    }
    & + label::before {
      transform: rotate(90deg) translateX(-3px);
    }
  }
`
export const CardContent = styled.div`
  ${({ theme }) => css`
  max-height: 0;
  overflow: hidden;
  transition: max-height .25s ease-in-out;
    p,
    span {
      font-size: ${theme.font.size.medium};
      font-weight: ${theme.font.medium};
      line-height: 19px;
    }
    p {
      color: ${theme.color.darkGray};
      margin-right: 9px;
      &:not(:last-child){
        margin-bottom: 16px;
      }
      &:first-child {
        margin-top: 24px;
      }
      &:last-child {
        margin-bottom: 24px;
      }
      span {
        color: ${theme.color.black};
        }
      }
    }
    .card-inner {
      button {
        margin: 0 auto;
      }
    }
  `}
`

export const Name = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.size.medium};
    font-weight: ${theme.font.medium};
    color: ${theme.color.black};
    line-height: 19px;
    margin-left: 20px;
    text-align: left;
    flex-grow: 1;
  `}
`
