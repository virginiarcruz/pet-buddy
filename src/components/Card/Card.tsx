import React from 'react'
import { CardContainer, Header, CardContent, Name, Input } from './styles'
import Avatar from 'components/Avatar'
import Button from 'components/Button'
import IconButton from 'components/IconButton'

export type CardProps = {
  petName?: string
  petType?: string
  petBreed?: string
  petGender?: string
  petColor?: string
  idCard: string
}

const Card = ({
  petName,
  petType,
  petBreed,
  petColor,
  petGender,
  idCard,
  ...props
}: CardProps) => (
  <CardContainer className="card-container">
    <Input type="checkbox" id={idCard} name="header-card" {...props} />
    <Header htmlFor={idCard} {...props}>
      <Avatar imageAlt="alt img" imageSrc="https://picsum.photos/50" />
      <Name> {petName} </Name>
      <IconButton />
    </Header>
    <CardContent className="card-content">
      <div className="card-inner">
        <p>
          Type: <span>{petType}</span>
        </p>
        <p>
          Breed: <span>{petBreed}</span>
        </p>
        <p>
          Gender: <span>{petGender}</span>
        </p>
        <p>
          Color: <span>{petColor}</span>
        </p>
        <Button>Details</Button>
      </div>
    </CardContent>
  </CardContainer>
)

export default Card
