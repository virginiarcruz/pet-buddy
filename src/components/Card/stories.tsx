import { Story, Meta } from '@storybook/react/types-6-0'
import Card from './Card'

export default {
  title: 'Card',
  component: Card,
  args: {
    petName: 'Summer',
    petType: 'Dog',
    petBreed: 'Abyssinian',
    petGender: 'Female',
    petColor: 'Black'
  }
} as Meta

export const Default: Story = (args) => (
  <Card
    imageAlt="description image"
    imageSrc="https://picsum.photos/50"
    idCard="id"
    {...args}
  />
)
