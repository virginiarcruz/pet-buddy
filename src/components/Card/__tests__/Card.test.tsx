import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Card from '../Card'

describe('<Card />', () => {
  it('should render the <Card />', () => {
    const { container } = renderWithTheme(
      <Card
        idCard="idCard"
        imageAlt="description image"
        imageSrc="https://picsum.photos/50"
        petBreed="Abyssinian"
        petColor="Black"
        petGender="Female"
        petName="Summer"
        petType="Dog"
      />
    )

    expect(container).toMatchSnapshot()

    expect(
      screen.getByRole('img', { name: /description image/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /summer/i })).toBeInTheDocument()
    expect(
      screen.getByRole('button', {
        name: /details/i
      })
    ).toBeInTheDocument
  })
})
