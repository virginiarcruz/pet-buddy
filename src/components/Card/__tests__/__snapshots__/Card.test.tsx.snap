// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the <Card /> 1`] = `
.c0 {
  background-color: #F8F8FC;
  border-radius: 12px;
  padding: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: none;
}

.c1:checked + label .icon-button {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c1:checked + label + .card-content {
  max-height: 100vh;
}

.c1:checked + label::before {
  -webkit-transform: rotate(90deg) translateX(-3px);
  -ms-transform: rotate(90deg) translateX(-3px);
  transform: rotate(90deg) translateX(-3px);
}

.c7 {
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height .25s ease-in-out;
  transition: max-height .25s ease-in-out;
}

.c7 p,
.c7 span {
  font-size: 16px;
  font-weight: 500;
  line-height: 19px;
  text-align: left;
}

.c7 p {
  color: #707183;
  margin-right: 9px;
  margin-bottom: 16px;
}

.c7 p:first-child {
  margin-top: 24px;
}

.c7 p:nth-child(4) {
  margin-bottom: 24px;
}

.c7 p span {
  color: #000121;
}

.c7 .card-inner button {
  margin: 0 auto;
}

.c4 {
  font-size: 16px;
  font-weight: 500;
  color: #000121;
  line-height: 19px;
  margin-left: 20px;
  text-align: left;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  border-radius: 100px;
  width: 40px;
  height: 40px;
  cursor: pointer;
}

.c8 {
  border: 0;
  border-radius: 8px;
  background-color: #002FD6;
  color: #FFFFFF;
  font-family: 'Inter',-apple-system,sans-serif;
  font-size: 16px;
  font-weight: 600;
  line-height: 19px;
  padding: 16px 99px;
  max-width: 263px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  margin-left: 4px;
  width: 7px;
  height: 12px;
}

.c9 path {
  fill: #FFFFFF;
}

.c5 {
  border-radius: 100px;
  background-color: #002FD6;
  color: #FFFFFF;
  width: 24px;
  height: 24px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  width: 7px;
  height: 12px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c6 path {
  fill: #FFFFFF;
}

@media (min-width:768px) {
  .c3 {
    width: 48px;
    height: 48px;
  }
}

<div>
  <section
    class="c0 card-container"
  >
    <input
      class="c1"
      id="idCard"
      name="header-card"
      type="checkbox"
    />
    <label
      class="c2"
      for="idCard"
    >
      <img
        alt="description image"
        class="c3"
        src="https://picsum.photos/50"
      />
      <h2
        class="c4"
      >
         
        Summer
         
      </h2>
      <div
        class="c5 icon-button"
      >
        <svg
          class="c6"
          viewBox="0 0 7 12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.821 6.454a.665.665 0 0 0 0-.908L1.687.186c-.264-.248-.661-.248-.9 0L.18.818c-.239.247-.239.66 0 .935l4.075 4.233L.18 10.247c-.239.275-.239.688 0 .935l.608.632c.239.248.636.248.9 0l5.134-5.36z"
          />
        </svg>
      </div>
    </label>
    <div
      class="c7 card-content"
    >
      <div
        class="card-inner"
      >
        <p>
          Type: 
          <span>
            Dog
          </span>
        </p>
        <p>
          Breed: 
          <span>
            Abyssinian
          </span>
        </p>
        <p>
          Gender: 
          <span>
            Female
          </span>
        </p>
        <p>
          Color: 
          <span>
            Black
          </span>
        </p>
        <button
          class="c8"
        >
          <span>
            Details
          </span>
          <svg
            class="c9"
            viewBox="0 0 7 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.821 6.454a.665.665 0 0 0 0-.908L1.687.186c-.264-.248-.661-.248-.9 0L.18.818c-.239.247-.239.66 0 .935l4.075 4.233L.18 10.247c-.239.275-.239.688 0 .935l.608.632c.239.248.636.248.9 0l5.134-5.36z"
            />
          </svg>
        </button>
      </div>
    </div>
  </section>
</div>
`
